From stern@rowland.harvard.edu Tue Dec 20 07:01:47 2005
Date: Tue, 20 Dec 2005 09:58:08 -0500 (EST)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>
Subject: [PATCH] UHCI: Don't log short transfers
Message-ID: <Pine.LNX.4.44L0.0512200953140.5250-100000@iolanthe.rowland.org>

Even when the URB_SHORT_NOT_OK flag is set, a short transfer shouldn't
generate a debugging log message.  Especially not one with the confusing
claim that the transfer "failed with status 0".  This patch (as627)
fixes that behavior in uhci-hcd.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/uhci-q.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/host/uhci-q.c
+++ gregkh-2.6/drivers/usb/host/uhci-q.c
@@ -922,7 +922,6 @@ static int uhci_result_common(struct uhc
 td_error:
 	ret = uhci_map_status(status, uhci_packetout(td_token(td)));
 
-err:
 	if ((debug == 1 && ret != -EPIPE) || debug > 1) {
 		/* Some debugging code */
 		dev_dbg(uhci_dev(uhci), "%s: failed with status %x\n",
@@ -934,6 +933,7 @@ err:
 			lprintk(errbuf);
 		}
 	}
+err:
 
 	/* Note that the queue has stopped and save the next toggle value */
 	urbp->qh->element = UHCI_PTR_TERM;
